Assets/Scripts/AbilityLibrary.cs

```diff
@@ -0,0 +1,126 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+public class AbilityLibrary : MonoBehaviour
+{
+
+    private GameObject player;
+    private Animator animator;
+    [SerializeField] Destructible target;   //ned to generalize, see the thing below
+
+    [SerializeField] string[] names;
+    [SerializeField] Image[] UIimages;
+    [SerializeField] TMP_Text[] UItexts;
+
+    Dictionary<string, Ability> abilites = new Dictionary<string, Ability>();
+    Ability slice, dizzy, unsurprisingSlash, fireSpell;        //might need to turn this into an array later if it gets out of control
+
+    [SerializeField] GameObject defaultSpellBallPrefab;
+
+    //TO ADD AN ABILITY
+    //1. add the name to the line directly above this
+    //2. define it in start
+    //3. add it to the abilities list in start with the addAbility() method
+    //4. add an if statement in update and useAbility(), ie if (condition1 && condition2 && attack1.useAbility()), the useAbility method will manage the cooldown, the if statement will not be entered if the ability is not cooled down
+    //5. add any stuff you want to the if statement body (effects, extra damage, etc), the use ability will handle the base damage, base animation, and the cooldown UI
+    //6. add the name and UI to the appropriate list fields of this script in the Unity Inspector Window
+    //make sure everything is spelled correctly, make sure to use double ands (&&) in the if loops
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        player = gameObject;
+        animator = player.GetComponent<Animator>();
+        if (UIimages.Length != UItexts.Length) { Debug.Log("Ability UI array length mismatch"); }
+
+        //abilites
+        slice = new Ability("slice", player, 10, 10, "coolAttack");
+        dizzy = new Ability("dizzy", player, 2, 20, "");
+        unsurprisingSlash = new Ability("unsurprisingSlash", player, 1, 6, "aa");
+        fireSpell = new Ability("fireSpell", player, 0, 3, "aa");
+
+
+        //add abilities to the abilities list
+        addAbility(slice);
+        addAbility(dizzy);
+        addAbility(unsurprisingSlash);
+        addAbility(fireSpell);
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+        Enemy enemy = target.GetComponent<Enemy>(); //hopefully this whole target and enemy thing can be generalized, I will get to this later though
+        //actually in order to get this to be more general to other enemies/multiple enimies/hitboxes/effects/special damage, I think the ability might just store the cooldown and everything else can be done in the if, we'll see
+
+        //slice
+        if (Input.GetKey("1") && slice.useAbility(target)) { }
+
+        //dizzy
+        if (Input.GetKey("2") && dizzy.useAbility(target))
+        {
+            enemy.MakeDizzy();
+        }
+
+        //unsurprising
+        if (Input.GetKey("3") && Vector3.Distance(transform.position, target.gameObject.transform.position) <= 4
```

Comment:
With how much things can change with abilties and whatnot, and with the fact that other abilities will likely use a range indicator, constants could get messy and declaring a range variable might be smarter

Reply:
I'm going to fix by adding two things. method inRange(double range) will shorten the if statements by doing that Vector3.distance stuff. Then I'll add the variables. So the command inRange(meleeRange) can be used in the if statements

Reply:
fixed